src/app/components/searchInput/searchForm.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+export default class SearchInput extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			searchQuery: '',
+		};
+		this.searchInput = React.createRef();
+		this.onKeyUp = this.onKeyUp.bind(this);
+	}
+
+	onKeyUp(e) {
+		const { mode } = this.props;
+		if (mode === 'Immediate') {
+			const searchQuery = this.searchInput.current.value;
+			this.setState({ searchQuery });
+		} else if (mode === 'onEnterClick') {
+			if (e.keyCode === 13) {
+				const searchQuery = this.searchInput.current.value;
+				this.setState({ searchQuery });
+			}
+		} else if (mode === 'onStopTyping') {
+			const doneTyping = () => {
+				const searchQuery = this.searchInput.current.value;
+				this.setState({ searchQuery });
+			};
+			let typingTimer;
+			if (typingTimer) {
+				clearTimeout(typingTimer);
+			}
+			if (this.searchInput.current.value) {
+				typingTimer = setTimeout(doneTyping, 1000);
+			}
+		}
```

Comment:
I believe it'll be prettier to use switch construction in such situations (3 and more choose cases)